### DISCUSSION TITLE
`engines` for Electron targets

### DISCUSSION THREAD

**Discussion by goosewobbler** [parcel-bundler/parcel#8666] [Q&A] [answered]
Hi,

We're using Parcel 2 to build Electron targets, I'm just wondering what the best value to put in `engines` is.  If we specify `electron` here will it have an effect?  Basically is the below proposal implemented yet:

https://github.com/parcel-bundler/parcel/issues/2492#issuecomment-450999156

Or is it just currently ignored, and what is the behaviour in this case?  Our app works fine specifying electron 22 there, combined with `browserslist: "last 1 Electron major version"` but could do with clarification on what Parcel is doing.

**Comment by mischnic** [comment, marked as answer]
With `--log-level verbose`, you should be able to see what exactly Parcel has detected: https://github.com/parcel-bundler/parcel/pull/8254

With `browserslist`, Parcel either targets node or the browser (context=node or =browser), and with electron inside `engines`, it would also set the electron context, but that doesn't do much right now:

The only electron-specific code currently is https://github.com/parcel-bundler/parcel/blob/ae31c3c85ff2a3f8353d5c40928d46d9bc3a0545/packages/utils/node-resolver-core/src/NodeResolver.js#L775-L777
which makes sure Parcel doesn't try to resolve an import to `electron`, but  I think only applies for the electron renderer process.

> Basically is the below proposal implemented yet:

No

**Reply by goosewobbler** [reply]
Thanks for the comprehensive answer!
